### ISSUE TITLE
NPE org.springframework.ide.vscode.commons.jdtls.Wrappers.wrap(Wrappers.java:55)

### ISSUE THREAD

**Issue by alpapad** [spring-projects/sts4#471]
00:08:21.109 [Reconciler-1] INFO  o.s.ide.vscode.commons.util.Log - Indexed 1747 properties.
00:08:21.110 [Reconciler-1] ERROR o.s.i.v.boot.metadata.types.TypeUtil - 
java.lang.NullPointerException: null
	at org.springframework.ide.vscode.commons.jdtls.Wrappers.wrap(Wrappers.java:55)
	at org.springframework.ide.vscode.commons.jdtls.Wrappers$2.component(Wrappers.java:117)
	at org.springframework.ide.vscode.boot.metadata.types.Type.fromJavaType(Type.java:142)
	at org.springframework.ide.vscode.boot.metadata.types.TypeUtil.lambda$getProperties$5(TypeUtil.java:769)
	at java.base/java.util.stream.ReferencePipeline$2$1.accept(ReferencePipeline.java:177)
	at java.base/java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:195)
	at java.base/java.util.ArrayList$ArrayListSpliterator.forEachRemaining(ArrayList.java:1654)
	at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:484)
	at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:474)
	at java.base/java.util.stream.StreamSpliterators$WrappingSpliterator.forEachRemaining(StreamSpliterators.java:312)
	at java.base/java.util.stream.Streams$ConcatSpliterator.forEachRemaining(Streams.java:734)
	at java.base/java.util.stream.ReferencePipeline$Head.forEach(ReferencePipeline.java:658)
	at org.springframework.ide.vscode.boot.metadata.types.TypeUtil.getProperties(TypeUtil.java:765)
	at org.springframework.ide.vscode.boot.metadata.types.TypeUtil.getPropertiesMap(TypeUtil.java:914)
	at org.springframework.ide.vscode.boot.yaml.reconcile.ApplicationYamlASTReconciler.reconcile(ApplicationYamlASTReconciler.java:264)
	at org.springframework.ide.vscode.boot.yaml.reconcile.ApplicationYamlASTReconciler.reconcile(ApplicationYamlASTReconciler.java:221)
	at org.springframework.ide.vscode.boot.yaml.reconcile.ApplicationYamlASTReconciler.reconcile(ApplicationYamlASTReconciler.java:281)
	at org.springframework.ide.vscode.boot.yaml.reconcile.ApplicationYamlASTReconciler.reconcile(ApplicationYamlASTReconciler.java:221)
	at org.springframework.ide.vscode.boot.yaml.reconcile.ApplicationYamlASTReconciler.reconcile(ApplicationYamlASTReconciler.java:297)
	at org.springframework.ide.vscode.boot.yaml.reconcile.ApplicationYamlASTReconciler.reconcile(ApplicationYamlASTReconciler.java:218)
	at org.springframework.ide.vscode.boot.yaml.reconcile.ApplicationYamlASTReconciler.reconcile(ApplicationYamlASTReconciler.java:281)
	at org.springframework.ide.vscode.boot.yaml.reconcile.ApplicationYamlASTReconciler.reconcile(ApplicationYamlASTReconciler.java:221)
	at org.springframework.ide.vscode.boot.yaml.reconcile.ApplicationYamlASTReconciler.reconcile(ApplicationYamlASTReconciler.java:297)
	at org.springframework.ide.vscode.boot.yaml.reconcile.ApplicationYamlASTReconciler.reconcile(ApplicationYamlASTReconciler.java:218)
	at org.springframework.ide.vscode.boot.yaml.reconcile.ApplicationYamlASTReconciler.reconcile(ApplicationYamlASTReconciler.java:184)
	at org.springframework.ide.vscode.boot.yaml.reconcile.ApplicationYamlASTReconciler.reconcile(ApplicationYamlASTReconciler.java:110)
	at org.springframework.ide.vscode.boot.yaml.reconcile.ApplicationYamlASTReconciler.reconcile(ApplicationYamlASTReconciler.java:189)
	at org.springframework.ide.vscode.boot.yaml.reconcile.ApplicationYamlASTReconciler.reconcile(ApplicationYamlASTReconciler.java:110)
	at org.springframework.ide.vscode.boot.yaml.reconcile.ApplicationYamlASTReconciler.reconcile(ApplicationYamlASTReconciler.java:98)
	at org.springframework.ide.vscode.boot.yaml.reconcile.ApplicationYamlASTReconciler.reconcile(ApplicationYamlASTReconciler.java:91)
	at org.springframework.ide.vscode.commons.yaml.reconcile.YamlReconcileEngine.reconcile(YamlReconcileEngine.java:45)
	at org.springframework.ide.vscode.boot.properties.BootPropertiesLanguageServerComponents.lambda$getReconcileEngine$7(BootPropertiesLanguageServerComponents.java:184)
	at org.springframework.ide.vscode.commons.languageserver.util.SimpleLanguageServer.lambda$validateWith$11(SimpleLanguageServer.java:613)
	at reactor.core.publisher.MonoRunnable.subscribe(MonoRunnable.java:49)
	at reactor.core.publisher.Mono.subscribe(Mono.java:4210)
	at reactor.core.publisher.MonoSubscribeOn$SubscribeOnSubscriber.run(MonoSubscribeOn.java:124)
	at reactor.core.scheduler.WorkerTask.call(WorkerTask.java:84)
	at reactor.core.scheduler.WorkerTask.call(WorkerTask.java:37)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at java.base/java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:304)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at java.base/java.lang.Thread.run(Thread.java:834)

**Comment by BoykoAlex** [comment]
Looks important to fix! Any chance you can provide more content for the issue?
The stack trace seem to indicate that:
1. Spring Boot properties YAML file is opened (any chance you can attach it)
2. Reconciling fails for a property that is Array of some objects... 

Regarding 2. Any idea which property might be causing this exception? Maybe try commenting out a few candidate properties+values and check if exception stops showing up. Let us know please which property that is if you'd be able to find this out. Otherwise lets see if you could post your YAML file here.

**Comment by BoykoAlex** [comment]
@alpapad any update? I'd very very interested to take a look at your boot properties YAML file.

**Comment by didorg** [comment]
> 00:08:21.109 [Reconciler-1] INFO o.s.ide.vscode.commons.util.Log - Indexed 1747 properties.
> 00:08:21.110 [Reconciler-1] ERROR o.s.i.v.boot.metadata.types.TypeUtil -
> java.lang.NullPointerException: null
> at org.springframework.ide.vscode.commons.jdtls.Wrappers.wrap(Wrappers.java:55)
> at org.springframework.ide.vscode.commons.jdtls.Wrappers$2.component(Wrappers.java:117)
> at org.springframework.ide.vscode.boot.metadata.types.Type.fromJavaType(Type.java:142)
> at org.springframework.ide.vscode.boot.metadata.types.TypeUtil.lambda$getProperties$5(TypeUtil.java:769)
> at java.base/java.util.stream.ReferencePipeline$2$1.accept(ReferencePipeline.java:177)
> at java.base/java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:195)
> at java.base/java.util.ArrayList$ArrayListSpliterator.forEachRemaining(ArrayList.java:1654)
> at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:484)
> at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:474)
> at java.base/java.util.stream.StreamSpliterators$WrappingSpliterator.forEachRemaining(StreamSpliterators.java:312)
> at java.base/java.util.stream.Streams$ConcatSpliterator.forEachRemaining(Streams.java:734)
> at java.base/java.util.stream.ReferencePipeline$Head.forEach(ReferencePipeline.java:658)
> at org.springframework.ide.vscode.boot.metadata.types.TypeUtil.getProperties(TypeUtil.java:765)
> at org.springframework.ide.vscode.boot.metadata.types.TypeUtil.getPropertiesMap(TypeUtil.java:914)
> at org.springframework.ide.vscode.boot.yaml.reconcile.ApplicationYamlASTReconciler.reconcile(ApplicationYamlASTReconciler.java:264)
> at org.springframework.ide.vscode.boot.yaml.reconcile.ApplicationYamlASTReconciler.reconcile(ApplicationYamlASTReconciler.java:221)
> at org.springframework.ide.vscode.boot.yaml.reconcile.ApplicationYamlASTReconciler.reconcile(ApplicationYamlASTReconciler.java:281)
> at org.springframework.ide.vscode.boot.yaml.reconcile.ApplicationYamlASTReconciler.reconcile(ApplicationYamlASTReconciler.java:221)
> at org.springframework.ide.vscode.boot.yaml.reconcile.ApplicationYamlASTReconciler.reconcile(ApplicationYamlASTReconciler.java:297)
> at org.springframework.ide.vscode.boot.yaml.reconcile.ApplicationYamlASTReconciler.reconcile(ApplicationYamlASTReconciler.java:218)
> at org.springframework.ide.vscode.boot.yaml.reconcile.ApplicationYamlASTReconciler.reconcile(ApplicationYamlASTReconciler.java:281)
> at org.springframework.ide.vscode.boot.yaml.reconcile.ApplicationYamlASTReconciler.reconcile(ApplicationYamlASTReconciler.java:221)
> at org.springframework.ide.vscode.boot.yaml.reconcile.ApplicationYamlASTReconciler.reconcile(ApplicationYamlASTReconciler.java:297)
> at org.springframework.ide.vscode.boot.yaml.reconcile.ApplicationYamlASTReconciler.reconcile(ApplicationYamlASTReconciler.java:218)
> at org.springframework.ide.vscode.boot.yaml.reconcile.ApplicationYamlASTReconciler.reconcile(ApplicationYamlASTReconciler.java:184)
> at org.springframework.ide.vscode.boot.yaml.reconcile.ApplicationYamlASTReconciler.reconcile(ApplicationYamlASTReconciler.java:110)
> at org.springframework.ide.vscode.boot.yaml.reconcile.ApplicationYamlASTReconciler.reconcile(ApplicationYamlASTReconciler.java:189)
> at org.springframework.ide.vscode.boot.yaml.reconcile.ApplicationYamlASTReconciler.reconcile(ApplicationYamlASTReconciler.java:110)
> at org.springframework.ide.vscode.boot.yaml.reconcile.ApplicationYamlASTReconciler.reconcile(ApplicationYamlASTReconciler.java:98)
> at org.springframework.ide.vscode.boot.yaml.reconcile.ApplicationYamlASTReconciler.reconcile(ApplicationYamlASTReconciler.java:91)
> at org.springframework.ide.vscode.commons.yaml.reconcile.YamlReconcileEngine.reconcile(YamlReconcileEngine.java:45)
> at org.springframework.ide.vscode.boot.properties.BootPropertiesLanguageServerComponents.lambda$getReconcileEngine$7(BootPropertiesLanguageServerComponents.java:184)
> at org.springframework.ide.vscode.commons.languageserver.util.SimpleLanguageServer.lambda$validateWith$11(SimpleLanguageServer.java:613)
> at reactor.core.publisher.MonoRunnable.subscribe(MonoRunnable.java:49)
> at reactor.core.publisher.Mono.subscribe(Mono.java:4210)
> at reactor.core.publisher.MonoSubscribeOn$SubscribeOnSubscriber.run(MonoSubscribeOn.java:124)
> at reactor.core.scheduler.WorkerTask.call(WorkerTask.java:84)
> at reactor.core.scheduler.WorkerTask.call(WorkerTask.java:37)
> at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
> at java.base/java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:304)
> at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
> at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
> at java.base/java.lang.Thread.run(Thread.java:834)

same problem

**Comment by BoykoAlex** [comment]
@didorg can you please see if you can attach your YAML or project here? Any idea which property in the YAML cause the exception?

**Comment by didorg** [comment]
Steps:
1. Install Eclipse 2020-06 (4.16.0) [macOs catalina, version 10.15.5]
2. Eclipse Marketplace -> Spring Tools 4 (aka Spring tool Suite 4) 4.7.0.RELEASE -> **Install** - **Restart**
3. I do not need to create or open any project to reproduce this error. After the second step. I can't open any more eclipse. I always get this error

[error.log](https://github.com/spring-projects/sts4/files/4822671/error.log)

<img width="923" alt="Screen Shot 2020-06-23 at 9 12 22 PM" src="https://user-images.githubusercontent.com/22514083/85485214-42e13a00-b596-11ea-961b-ea085830ab03.png">

**Comment by martinlippert** [comment]
@didorg The error log reads like there is something going wrong while initializing the dark theme in the Eclipse platform. What happens if you open an empty, fresh workspace with the same IDE installation? Does that work?

In case that works, you could try to switch that new fresh workspace to dark mode and see if that reproduces the issue. That way, we would have a reproducible path that works from a fresh install without specific workspace settings.

And what happens if you download the ready-to-use distribution of Spring Tools 4 for Eclipse from https://spring.io/tools? It contains Eclipse 2020-06 as base platform, too. Does that behave differently?

**Comment by didorg** [comment]
The ready-to-use distribution of Spring Tools 4 for Eclipse from https://spring.io/tools works perfectly. Thanks!

**Comment by martinlippert** [comment]
Glad to hear that and thanks for letting us know.

**Comment by BoykoAlex** [comment]
@alpapad @didorg The NPE coming from:
```
java.lang.NullPointerException: null
at org.springframework.ide.vscode.commons.jdtls.Wrappers.wrap(Wrappers.java:55)
at org.springframework.ide.vscode.commons.jdtls.Wrappers$2.component(Wrappers.java:117)
at org.springframework.ide.vscode.boot.metadata.types.Type.fromJavaType(Type.java:142)
at org.springframework.ide.vscode.boot.metadata.types.TypeUtil.lambda$getProperties$5(TypeUtil.java:769)
```
Should be fixed with 51b6406222b1d9803482b99693e8bce5ec592e71
